src/OpenSpaceToolkit/Physics/Data/Manager.cpp

```diff
@@ -180,8 +180,8 @@ void Manager::checkManifestAgeAndUpdate_() const
 
     const Duration manifestAge = Instant::Now() - manifest_.getLastModifiedTimestamp();
 
-    // If loaded manifest is old enough, fetch a new one and load it.
-    if (nextUpdateCheckTimestamp < manifest_.getLastModifiedTimestamp() && manifestAge > DataRefreshRate_())
+    // If current time is past next update check time or manifest is old enough, fetch a new one and load it.
+    if (Instant::Now() > nextUpdateCheckTimestamp || manifestAge > DataRefreshRate_())
```

Comment:
Should this remain an `AND` condition? so that `DataRefreshRate_()` remains as a "maximum" frequency. Also, this means we won't unnecessarily check this every `DataRefreshRate_()`, when we haven't yet reached the `nextUpdateCheckTimestamp` (and therefore we "know" there isn't any new data to fetch)

```suggestion
    if (Instant::Now() > nextUpdateCheckTimestamp && manifestAge > DataRefreshRate_())
```